scene/gui/line_edit.cpp

```diff
@@ -2281,7 +2284,15 @@ void LineEdit::_shape() {
 	if (text.length() == 0 && ime_text.length() == 0) {
 		t = placeholder_translated;
 	} else if (pass) {
-		t = secret_character.repeat(text.length() + ime_text.length());
+		// TODO: Integrate with text server to add support for non-latin scripts.
+		// Allow secret_character as empty strings, act like if a space was used as a secret character.
+		String secret = " ";
+		// Allow values longer than 1 character in the property, but trim characters after the first one.
+		if (!secret_character.is_empty()) {
+			secret = secret_character.left(1);
+			update_configuration_warnings();
```

Comment:
This call should be in `set_secret_character()`, it shouldn't be called excessively.

Reply:
Agreed, thanks.